tests/test_config_io.py

```
@@ -0,0 +1,323 @@
+    assert set(d.keys()) == {"model", "subagents", "middleware", "knowledge"}
+    assert d["model"]["name"] == cfg.model_name
+    assert d["model"]["temperature"] == cfg.temperature
+    assert d["subagents"]["worker"]["tools"] == list(cfg.worker.tools)
+    assert d["middleware"]["audit"] == cfg.audit_middleware
+    assert d["knowledge"]["top_k"] == cfg.knowledge_top_k
+
+
+# ── validate_config_dict ─────────────────────────────────────────────────────
+
+
+@pytest.mark.parametrize("bad_value,expected_error_fragment", [
```

Ruff (0.15.10)

[warning] Wrong type passed to first argument of `pytest.mark.parametrize`; expected `tuple`

Use a `tuple` for the first argument

(PT006)